nl_server/config.py

```diff
@@ -48,54 +46,55 @@ class ModelUsage(str, Enum):
   RERANKING = 'RERANKING'
 
 
-@dataclass
-class ModelConfig(ABC):
-  type: str
-  score_threshold: float
-  usage: str
+@dataclass(kw_only=True)
+class ModelConfig:
+  type: str = None
+  usage: str = None
+  score_threshold: float = None
 
 
-@dataclass
+@dataclass(kw_only=True)
 class VertexAIModelConfig(ModelConfig):
-  project_id: str
-  location: str
-  prediction_endpoint_id: str
+  project_id: str = None
+  location: str = None
+  prediction_endpoint_id: str = None
 
 
-@dataclass
+@dataclass(kw_only=True)
 class LocalModelConfig(ModelConfig):
-  gcs_folder: str = ''
+  gcs_folder: str = None
 
 
-@dataclass
-class IndexConfig(ABC):
-  store_type: str
-  model: str
-  healthcheck_query: str
+@dataclass(kw_only=True)
+class IndexConfig:
+  store_type: str = None
+  model: str = None
+  healthcheck_query: str = 'health'
 
 
-@dataclass
+@dataclass(kw_only=True)
 class MemoryIndexConfig(IndexConfig):
-  embeddings_path: str
+  embeddings_path: str = None
 
 
-@dataclass
+@dataclass(kw_only=True)
 class LanceDBIndexConfig(IndexConfig):
-  embeddings_path: str
+  embeddings_path: str = None
 
 
-@dataclass
+@dataclass(kw_only=True)
 class VertexAIIndexConfig(IndexConfig):
-  project_id: str
-  location: str
-  index_endpoint_root: str
-  index_endpoint: str
-  index_id: str
+  project_id: str = None
+  location: str = None
+  index_endpoint_root: str = None
+  index_endpoint: str = None
+  index_id: str = None
 
 
-# Defines one embeddings index config.
-@dataclass
-class EmbeddingsConfig:
+# This represents the full catalog of models and indexes.
```

Comment:
... a subset of them may not be enabled by the Env.

Reply:
Done